src/main/java/at/hannibal2/skyhanni/features/itemabilities/abilitycooldown/ItemAbility.kt

```diff
@@ -128,6 +132,22 @@ enum class ItemAbility(
             return entries.firstOrNull { it.newVariant && internalName in it.internalNames }
         }
 
+        fun getAllAbilityScrolls(itemStack: ItemStack?): List<ItemAbility> {
+            val list = mutableListOf<ItemAbility>()
+            for (ability in ItemAbility.entries) {
+                for (scroll in itemStack?.getAbilityScrolls().orEmpty()) {
```

Comment:
do not fetch the list of ability scrolls once per item ability. rather do this once outside the for loop and reuse the variable.